### PR TITLE
Oracle crawler

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

To support Oracle database.

### 🤓 What?

- Refactor the MySQL extractor into a generic extractor that uses SQLAlchemy to get the metadata.
- Basic table/view/materialized view information
- table/column comment
- DDL
- query history but Oracle only keeps the history for a very short time

### 🧪 Tested?

End-to-end test on dev. (Only the ingestion part)

Unit tests added

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/oracle/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Oracle Connector
+
+This connector extracts technical metadata from a MySQL database using [oracledb](https://oracle.github.io/python-oracledb/) library.
+
+## Setup
+
+We recommend you to create a role
```
Comment (mars-lan) [inline metaphor/oracle/README.md:7]: Complete the sentence?

File: metaphor/oracle/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+-- Step 2: Create the user
+CREATE USER metaphor_user IDENTIFIED BY my_password;
+
+-- Step 3: Assign the role to the user
+GRANT metaphor TO metaphor_user;
+
+```
+
+We also need `SELECT` privileges on table and views to get metadata
```
Comment (mars-lan) [inline metaphor/oracle/README.md:25]: Can doc we can point to?

File: metaphor/oracle/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+### Optional Configurations
+
+By default, the connector will connect using the default MySQL port 3306. You can change it using the following config:
+
+```yaml
+port: <port_number>
+```
+
+If you are indirectly connect to the database, you need to set the real host of the database using the following config:
```
Comment (mars-lan) [inline metaphor/oracle/README.md:61]: Not sure I understand the use of this config?

File: metaphor/oracle/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# Oracle Connector
+
+This connector extracts technical metadata from a MySQL database using [oracledb](https://oracle.github.io/python-oracledb/) library.
```
Comment (mars-lan) [inline metaphor/oracle/README.md:3]: Oracle
```suggestion
This connector extracts technical metadata from an Oracle database using [oracledb](https://oracle.github.io/python-oracledb/) library.
```

File: metaphor/oracle/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+```yaml
+host: <hostname>  # the host the crawler will connect to
+user: <username>
+password: <password>
+database: <default_database_for_connections>
+```
+
+### Optional Configurations
+
+By default, the connector will connect using the default MySQL port 3306. You can change it using the following config:
```
Comment (mars-lan) [inline metaphor/oracle/README.md:55]:
```suggestion
By default, the connector will connect using the default Oracle port 1521. You can change it using the following config:
```

File: metaphor/oracle/README.md (hunk 2 above)
Comment (elic-eon) [reply metaphor/oracle/README.md:25]: https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/ALL_TABLES.html

> ALL_TABLES describes the relational tables accessible to the current user. To gather statistics for this view, use the DBMS_STATS package.

Not sure if "accessible" means select

File: metaphor/oracle/config.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+from metaphor.database.config import GenericDatabaseRunConfig
+
+
+@dataclass(config=ConnectorConfig)
+class OracleQueryLogConfig:
+    # Number of days back of query logs to fetch, if 0, don't fetch query logs
+    lookback_days: int = 1
+
+    # Query log filter to exclude certain usernames
+    excluded_usernames: Set[str] = field(default_factory=lambda: set())
```
Comment (usefulalgorithm) [inline metaphor/oracle/config.py:16]:
```suggestion
    excluded_usernames: Set[str] = field(default_factory=set)
```

File: metaphor/oracle/extractor.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+        self._database = config.database.lower()
+
+    async def extract(self) -> Collection[ENTITY_TYPES]:
+        logger.info(f"Fetching metadata from Oracle host {self._config.host}")
+
+        inspector = OracleExtractor.get_inspector(self._get_sqlalchemy_url())
+
+        system_users = self.get_system_users(inspector)
+        system_users.extend(["RDSADMIN".lower()])
```
Comment (usefulalgorithm) [inline metaphor/oracle/extractor.py:55]:
```suggestion
        system_users.extend([RDSADMIN_USER.lower()])
```

**Review by usefulalgorithm — COMMENTED** [review]
I think I understand the PR, looks other than these two nits!

File: metaphor/oracle/README.md (hunk 2 above)
Comment (elic-eon) [reply metaphor/oracle/README.md]: But I can't see tables belongs to a schema if I don't grant `SELECT` to crawler user

File: metaphor/oracle/README.md (hunk 3 above)
Comment (elic-eon) [reply metaphor/oracle/README.md:61]: Like account to overwrite. I think we have some similar issues, users may run the crawler on different network environment so the `alternative_host` can be specified to host won't change a log.

**Review by mars-lan — COMMENTED** [review]
LGTM at a high leve. @usefulalgorithm pls review in more details.

File: metaphor/oracle/README.md (hunk 3 above)
Comment (elic-eon) [reply metaphor/oracle/README.md]: Updated the doc

**Review by usefulalgorithm — APPROVED** [review]
go go go

**Pull request merged**